src/mame/nmk/ddealer.cpp

```diff
@@ -186,13 +112,68 @@ class ddealer_state : public driver_device
 	// splitted foreground area
 	tilemap_t  *m_fg_tilemap_left;
 	tilemap_t  *m_fg_tilemap_right;
-
-	// MCU sim related
-	int   m_respcount;
-	u8    m_input_pressed;
-	u16   m_coin_input;
+	
+	u8 m_bus_status;
 };
 
+void ddealer_state::machine_start()
+{
+	save_item(NAME(m_bus_status));
+}
+
+void ddealer_state::machine_reset()
+{
+	m_bus_status = 0x04;
+}
+
+void ddealer_state::mcu_port6_w(u8 data)
+{
+	// the actual mechanism is a little more complex, but these are written at the
+	// start and end of the take / release bus function
+	if (data == 0x08)
+	{
+		logerror("%s: mcu_port6_w 68k bus taken, 68k stopped (data %02x)\n", machine().describe_context(), data);
+		m_maincpu->set_input_line(INPUT_LINE_HALT, ASSERT_LINE);
+	}
+	else if (data == 0xb)
+	{
+		logerror("%s: mcu_port6_w 68k bus returned, 68k started (data %02x)\n", machine().describe_context(), data);
+		m_maincpu->set_input_line(INPUT_LINE_HALT, CLEAR_LINE);
+	}
+	else
+	{
+		logerror("%s: mcu_port6_w 68k bus access (data %02x)\n", machine().describe_context(), data);
+	}
+}
+
+u8 ddealer_state::mcu_port5_r()
+{
+	return m_screen->vpos() >> 2;
+}
+
+u8 ddealer_state::mcu_port6_r()
+{
+	// again this is simplified for now
+	m_bus_status ^= 0x04;
+	return m_bus_status;
+}
```

Comment:
Once again @galibert is there a clean way to do this?  Magic side effects of reading ports that aren’t strobed is a fairly obvious hack.